extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -0,0 +1,120 @@
+import * as path from 'path';
+import {
+	BookContributionProvider, BookContribution
+} from './bookExtensions';
+
+const localize = nls.loadMessageBundle();
+
+export function registerBooksWidget(bookContributionProvider: BookContributionProvider): void {
+	azdata.ui.registerModelViewProvider('books-widget', async (view) => {
+		const container = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '100%',
+			height: '100%',
+			alignItems: 'left'
+		}).component();
+		const bookslocationContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '270px',
+			height: '100%',
+			alignItems: 'left',
+			position: 'absolute'
+		}).component();
+
+		let books = bookContributionProvider.contributions.map(contribution => {
+			const bookRow = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row'
+			}).component();
+			const tsgbooklink = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+				label: contribution.name, //localize('troubleshootingBooks', 'Troubleshooting Book'),
+				title: contribution.name // localize('troubleshootingBooksTitle', 'Troubleshooting Book'),
+			}).component();
+			tsgbooklink.onDidClick(() => {
+				PromptForFolder(contribution);
+			});
+			bookRow.addItem(tsgbooklink, {
+				CSSStyles: {
+					'width': '100%',
+					'color': '#0078d4',
+					'text-decoration': 'underline',
+					'padding-top': '10px',
+					'text-align': 'left'
+				}
+			});
+			return bookRow;
+		});
+
+
+		container.addItems(books, {
+			CSSStyles: {
+				'padding-left': '10px',
+				'border-top': 'solid 1px #ccc',
+				'box-sizing': 'border-box',
+				'user-select': 'text'
+			}
+		});
+		bookslocationContainer.addItem(container, {
+			CSSStyles: {
+				'padding-top': '25px',
+				'padding-left': '5px'
+			}
+		});
+		await view.initializeModel(bookslocationContainer);
+	});
+}
+
+async function PromptForFolder(bookContribution: BookContribution) {
+	let filter = {
+		'All files': ['*']
+	};
+	let uris = await vscode.window.showOpenDialog({
+		filters: filter,
+		canSelectFiles: false,
+		canSelectMany: false,
+		canSelectFolders: true,
+		openLabel: localize('labePickFolder', 'Pick Folder')
+	});
+	if (uris && uris.length > 0) {
+		let pickedFolder = uris[0];
+		await saveBooksToFolder(pickedFolder, bookContribution);
+		await promptToReloadWindow(pickedFolder);
+	}
+}
+
+async function saveBooksToFolder(folderUri: vscode.Uri, bookContribution: BookContribution): Promise<void> {
+	// Get book contributions
+	if (bookContribution && folderUri) {
+		//remove folder if exists
+		await fs.removeSync(path.join(folderUri.path, bookContribution.name));
+		//copy them from the books extension:
+		await fs.copy(bookContribution.path, folderUri.path);
+	}
+}
+function promptToReloadWindow(folderUri: vscode.Uri): void {
+	const actionReload = localize('strReload', 'Reload');
+	const actionOpenNew = localize('strOpenNewInstance', 'Open new instance');
```

Comment:
1. `strOpenNewinstance` -> `prompt.openNewInstance`.
2. Let's use double-quotes for the 2nd parameter (again it's coding convention in our repo, we should have a hygiene rule for it).